theano/gpuarray/basic_ops.py

```diff
@@ -1171,7 +1175,13 @@ def c_code(self, node, name, inputs, outputs, sub):
                          "GpuReshape: given shape is of incorrect "
                          "length (%%d should be %%d).",
                          PyArray_DIM(%(shape)s, 0), %(params)s->ndim);
-            %(fail)s;
+            %(just_fail)s;
+        }
+
+        new_dims = (size_t*) malloc(sizeof(size_t) * %(params)s->ndim);
```

Comment:
Why this change? This would slow down computation as it convert a stack alloc to a malloc. What was the memory leak here?

Reply:
Not a memory leak here, but the previous code uses syntax `size_t array[variable];` (creating an array on stack with a dynamic size). It compiles with GCC, but it is not standard. I assume we try to have a valid C/C++ code.

But I can remove these changes.